mgnify_pipelines_toolkit/analysis/shared/study_summary_generator.py

```diff
@@ -0,0 +1,263 @@
+        type=str,
+        help="File containing successful analyses generated by the pipeline",
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        required=True,
+        type=str,
+        choices=["summarise", "merge"],
+        help="Mode to either summarise analyses (summarise), or merge multiple existing summaries (merge)",
+    )
+    parser.add_argument(
+        "-p", "--prefix", required=True, type=str, help="Prefix to output"
+    )
+
+    args = parser.parse_args()
+
+    indir = args.indir
+    runs = args.runs
+    mode = args.mode
+    prefix = args.prefix
+
+    return indir, runs, mode, prefix
+
+
+def get_tax_file(run_acc, analyses_dir, db_label):
+
+    tax_file = ""
+
+    db_path = Path(f"{analyses_dir}/{run_acc}/taxonomy-summary/{db_label}")
+
+    if db_path.exists():
+        if db_label in TAXDB_LABELS:
+            tax_file = Path(
+                f"{analyses_dir}/{run_acc}/taxonomy-summary/{db_label}/{run_acc}_{db_label}.txt"
+            )
+            if not tax_file.exists():
+                logging.error(
+                    f"DB path exists but file doesn't - exiting. Path: {tax_file}"
+                )
+                exit(1)
+
+            file_size = tax_file.stat().st_size
+            if (
+                file_size == 0
+            ):  # Pipeline can generate files that are empty for ITS DBs (UNITE and ITSoneDB),
+                # so need to skip those. Should probably fix that at some point
+                tax_file = ""
+        elif db_label in ASV_TAXDB_LABELS:
+            # ASV tax files could have up to two files, one for each amplified region (maximum two from the pipeline).
+            # So will need to handle this differently to closed-reference files
+            asv_tax_files = glob.glob(
+                f"{analyses_dir}/{run_acc}/taxonomy-summary/{db_label}/*.txt"
+            )
+            asv_tax_files = [
+                Path(file) for file in asv_tax_files if "concat" not in file
+            ]  # Have to filter out concatenated file if it exists
+
+            tax_file = asv_tax_files
+
+    # output will be a Path if closed-reference file, or a list if ASV file
+    return tax_file
+
+
+def parse_one_tax_file(run_acc, tax_file):
+
+    res_df = pd.DataFrame()
+
+    with open(tax_file, "r") as fr:
+        for line in fr:
+            line = line.strip()
+            temp_lst = line.split("\t")
+            curr_count = temp_lst[0]
+            curr_tax = ";".join(temp_lst[1:])
+
+            res_df.loc[curr_tax, run_acc] = curr_count
+
+    return res_df
+
+
+def generate_db_summary(db_label, tax_files, output_prefix):
+
+    if db_label in TAXDB_LABELS:
+        df_list = []
+
+        for run_acc, tax_file in tax_files.items():
+            df_list.append(parse_one_tax_file(run_acc, tax_file))
+
+        res_df = df_list[0]
+        for df in df_list[1:]:
+            res_df = res_df.join(df, how="outer")
+        res_df = res_df.fillna(0)
```

Comment:
Can we use `pd.concat(df_list).fillna(0)`?

Reply:
I prefer to have them separate, but I tend to dislike adding too many function calls on the same line so I think it's personal 😄

Reply:
Ah yep - sorry I mainly meant the use of `concat` instead of a loop of `join`s

Reply:
oh sorry I misunderstood completely! I just tried that - works great, just needed to also sort the index again after it 😄 🙇